### DISCUSSION TITLE
Do Ixy in V

### DISCUSSION THREAD

**Discussion by Immortalin** [vlang/v#11610] [Ideas]
https://github.com/ixy-languages/ixy-languages

**Comment by dumblob** [comment]
This is a very interesting observation. Especially:

>Ixy can already achieve a high performance with relatively low batch sizes of 32-64 because it is a full user space driver. Other user space packet processing frameworks like netmap that rely on a kernel driver need larger batch sizes of 512 and above to amortize the larger overhead of communicating with the driver in the kernel.

Thanks for the pointer @Immortalin.

On the other hand I don't see any connection to *V* - or am I missing something?

**Comment by gslicer** [comment]
> On the other hand I don't see any connection to _V_ - or am I missing something?

@dumblob I think he asks to implement this user-space driver in native V as part of the network library, so the list can be extended as this:

Language | Code | Status | Full evaluation

C | ixy.c | Finished | Paper
Rust | ixy.rs | Finished | Thesis, Rust vs. C performance comparison
Go | ixy.go | Finished | Thesis
C# | ixy.cs | Finished | Thesis
Java | ixy.java | Finished | (WIP), GC comparison
OCaml | ixy.ml | Finished | Documentation
Haskell | ixy.hs | Finished | Thesis
Swift | ixy.swift | Finished | Documentation
Javascript | ixy.js | Finished | (WIP)
Python | ixy.py | Finished | (WIP)
**V | ixy.v | WIP | (WIP)**

**Comment by Immortalin** [comment]
It would be a good and rigorous test of V's concurrency and IO performance in a real-world application.

**Comment by gslicer** [comment]
@Immortalin as long V emits C code that is beeing compiled, I would expect to perform somehwre between C and Rust - when V is able to compile & optimize to machine-code directly it might be indeed a good benchmark

**Comment by medvednikov** [comment]
> as long V emits C code that is beeing compiled, I would expect to perform somehwre between C and Rust

like the home page says, the performance is always going to be within 3% of C
